crates/rf24-rs/src/radio/rf24/mod.rs

```diff
@@ -40,6 +44,27 @@ pub enum Nrf24Error<SPI, DO> {
     NotAsTxError,
 }
 
+impl From<SpiError> for Nrf24Error<SpiError, OutputPinError> {
+    fn from(value: SpiError) -> Self {
+        Nrf24Error::Spi(value)
+    }
+}
+
+impl From<OutputPinError> for Nrf24Error<SpiError, OutputPinError> {
+    fn from(value: OutputPinError) -> Self {
+        Nrf24Error::Gpo(value)
+    }
+}
```

Comment:
These implementations cannot be covered with unit tests. Doing so would require changing the test harness to employ qemu, but that seems like overkill since

1. the emulated hardware would be infallible under a proper hardware setup
2. the qemu instance would need to interact with another qemu instance to emulate wireless transactions. I'm guessing this approach is not a thing we can do currently.

Reply:
I opened a couple issues to hopefully resolve this upstream in the mock library we're using for unit tests:
- dbrgn/embedded-hal-mock#128
- dbrgn/embedded-hal-mock#129 (would be considered a breaking change though)

Alternatively I can fork embedded-hal-mock and tell cargo to pull in my patched fork from git source instead of the original from crates.io.

Reply:
I have a locally stashed patch that will use my fork's patches (in my fork's dev branch). I'm waiting for feedback upstream before committing the patch. Local tests show these lines will be covered with these upstream patches

- https://github.com/dbrgn/embedded-hal-mock/pull/131
- https://github.com/dbrgn/embedded-hal-mock/pull/132